lib/streamlit/commands/query_params.py

```diff
@@ -142,3 +143,46 @@ def _ensure_no_embed_params(
         separator = "&" if current_embed_params else ""
         return separator.join([query_string, current_embed_params])
     return current_embed_params
+
+
+@dataclass
+class QueryParams:
+    """A dict-like representation of query params that generally behaves like st.session.state.
+    The main difference is that it only stores and returns str and list[str].
+
+    TODO(willhuang1997): Fill in these docs with examples and fix doc above. Above is just a stub for now.
+    """
+
+    query_params: Dict[str, List[Any]] = field(default_factory=dict)
```

Comment:
nit: this name should have a `_` prefix unless we expect users to access it directly, but I'm fairly certain we don't

Reply:
Should this class be `QueryParamsProxy`? I see `_UserInfoProxy` and `_SessionStateProxy` and it seems they use that proxy in order to make sure that users don't access this dictionary directly?

Reply:
Not entirely sure. I like the `SessionStateProxy` name to emphasize there's an additional layer of indirection between the backing dicts and the class users interact with, but I'm not a fan of the name `UserInfoProxy` since there's just one thin wrapper around a dict. This class is closer to the user info wrapper than session state.

We could either just name all of them `*Proxy` for consistency or change `UserInfoProxy` -> `UserInfo`. I prefer the latter, but it'd be worth asking @kajarenc about the `UserInfoProxy` name

Reply:
I think I prefer the latter too. I was looking at `SessionStateProxy` and I saw that we do filtering on "keyless" widgets but I was confused about `UserInfoProxy`. Thanks for the clarification. I'll leave it to be `QueryParams` instead of `QueryParamsProxy` as I don't think we really need any filtering.